File: SPOJ/6171.c
#include<stdio.h>
int main()
{
	int test_cases;
	scanf("%d",&test_cases);
	while(test_cases--)
	{
		int n,a[2002]={0},i,x,count,d=0,num;
		scanf("%d",&n);
		count = n/2 + 1;
		for(i=0;i<n;i++)
		{
			scanf("%d",&x);
			a[x+1000]++;
			if(a[x+1000] >= count)
			{
				d=1;
				num = x;
			}
		}
		if(d==1)
			printf("YES %d\n",num);
		else
			printf("NO\n");
	}
	return 0;
}

File: SPOJ/6171.py
t=input()
while t:
	t=t-1
	n=input()
	count = n/2 + 1
	y=raw_input().split()
	e={}
	d=1
	for i in y:
		if i in  e:
			e[i] = e[i] + 1
		else:
			e[i] = 1
	for i in e:
		if e[i] >= count:
			print "YES",i
			d=0
			break
	if d==1:
		print "NO"
